### ISSUE TITLE
Logback appender works fine in test but is not active in spring boot

### ISSUE THREAD

**Issue by MatthewDavidCampbell** [Microsoft/ApplicationInsights-Java#532]
Using 1.5.9 of Spring boot, Java 9 and the 2.0.BETA of the logback appender....

Registering the logback appender with a Spring `EnvironmentPostProcessor`  with the following code:

     LoggerContext logCtx = (LoggerContext) LoggerFactory.getILoggerFactory();

     // filter to restrict certain logger names (nothing special but does use logAlways)
     InsightsFilter filter = new InsightsFilter();
     filter.setContext(logCtx);
     filter.start();

     ApplicationInsightsAppender appender = new ApplicationInsightsAppender();
     appender.setInstrumentationKey(instrumentationKey);
     appender.setName(Constants.TELEMETRY_APPENDER_NAME);
     appender.setContext(logCtx);
     appender.addFilter(filter);
     appender.start();

     Logger log = logCtx.getLogger(Logger.ROOT_LOGGER_NAME);
     log.addAppender(appender);

     InternalLogger
            .INSTANCE
            .logAlways(
              InternalLogger.LoggingLevel.INFO, 
              "Activated Insights logging through logback", 
              new Object[0]
     );

This works fine in test and I can confirm sent messages to Insights plus that the InsightsFilter is used.  However, when running inside Spring-boot I see (thanks to the InternalLogger) that the appender is started but nothing happens against Insights.  It is like the appender is turned off.  Wondering if anybody else has had this problem?

**Comment by MatthewDavidCampbell** [comment]
Nothing wrong with the Insight stuff for Java just a miss on my part when registering the appender.  The EnvironmentPostProcessor was too early and when the context is Spring is being prepared the logging is redone.  So instead I added a Spring application event listener at ApplicationPreparedEvent.  Everything works fine now.

**Comment by dhaval24** [comment]
@MatthewDavidCampbell thanks for the update. Is it possible for you to share a snapshot of the complete code on SpringBoot 1.5.9 + Java 9 and AI with logback so others can also help fix if they get similar issue. Also do you think this is something worth to have on Microsoft docs / Github Wiki Page?

**Comment by MatthewDavidCampbell** [comment]
Problem I faced is partially due to not coding in Java for over 4 years and Spring which was then just getting started with Spring Boot.  Basically, from Spring 3 to Spring 5 there are "many" different hooks into the lifecycle of the Spring container.  I picked the wrong one not thinking about the implications.  The `ApplicatoinPreparedEvent` is what I should have picked since the logger is configured after the context refresh (? I think ?).  Plus I hate rigging logback with XML or YAML.  Wan't to start filter and appenders with code.  Sort of the same feeling with the ApplicationInsights.xml.  Like the setup with Serilog on the Net side I lot better where (for example) I can enrich the custom properties with an event filter.

Not sure a Wiki page is needed.  Made a duplicate question on Stack Overflow and posted my solution there.
